pkg/util/kms/kms_kmip_storage.go

```diff
@@ -0,0 +1,479 @@
+		return secrets.ErrSecretExists
+	}
+
+	// Register the key value the KMIP endpoint
+	value := plainText[secretID].(string)
+	valueBytes, err := base64.StdEncoding.DecodeString(value)
+	if err != nil {
+		return err
+	}
+
+	conn, err := k.connect()
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.PutSecret() can not connect %v", err)
+		return err
+	}
+	defer conn.Close()
+
+	registerPayload := kmip.RegisterRequestPayload{
+		ObjectType: kmip14.ObjectTypeSymmetricKey,
+		SymmetricKey: &kmip.SymmetricKey{
+			KeyBlock: kmip.KeyBlock{
+				KeyFormatType: kmip14.KeyFormatTypeRaw,
+				KeyValue: &kmip.KeyValue{
+					KeyMaterial: valueBytes,
+				},
+				CryptographicLength:    cryptographicLength,
+				CryptographicAlgorithm: kmip14.CryptographicAlgorithmAES,
+			},
+		},
+	}
+	registerPayload.TemplateAttribute.Append(kmip14.TagCryptographicUsageMask, kmip14.CryptographicUsageMaskExport)
+	respMsg, decoder, uniqueBatchItemID, err := k.send(conn, kmip14.OperationRegister, registerPayload)
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.PutSecret() can send %v", err)
+		return err
+	}
+	bi, err := k.response(respMsg, kmip14.OperationRegister, uniqueBatchItemID)
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.PutSecret() can verify response %v", err)
+		return err
+	}
+
+	var registerRespPayload kmip.RegisterResponsePayload
+	err = decoder.DecodeValue(&registerRespPayload, bi.ResponsePayload.(ttlv.TTLV))
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.PutSecret() can decode response payload %v", err)
+		return err
+	}
+
+	k.secret.StringData[KMIPUniqueID] = registerRespPayload.UniqueIdentifier
+	if !util.KubeUpdate(k.secret) {
+		log.Errorf("Failed to update KMS secret %v in ns %v", k.secret.Name, k.secret.Namespace)
+		return fmt.Errorf("Failed to update KMS secret %v in ns %v", k.secret.Name, k.secret.Namespace)
+	}
+	return nil
+}
+
+// DeleteSecret deletes the secret data associated with the
+// supplied secretId.
+func (k *KMIPSecretStorage) DeleteSecret(
+	secretID string,
+	keyContext map[string]string,
+) error {
+	log := util.Logger()
+
+	// Find the key ID
+	uniqueIdentifier, exists := k.secret.StringData[KMIPUniqueID]
+	if !exists {
+		log.Errorf("KMIPSecretStorage.DeleteSecret() No uniqueIdentifier in the secret")
+		return secrets.ErrInvalidSecretId
+	}
+
+	conn, err := k.connect()
+	if err != nil {
+		return err
+	}
+	defer conn.Close()
+
+	respMsg, decoder, uniqueBatchItemID, err := k.send(conn, kmip14.OperationDestroy, kmip.DestroyRequestPayload{
+		UniqueIdentifier: uniqueIdentifier,
+	})
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.DeleteSecret() can not send %v", err)
+		return err
+	}
+	bi, err := k.response(respMsg, kmip14.OperationDestroy, uniqueBatchItemID)
+	if err != nil {
+		log.Errorf("KMIPSecretStorage.DeleteSecret() can verify respnse %v", err)
+		return err
+	}
+
+	var destroyRespPayload kmip.DestroyResponsePayload
+	err = decoder.DecodeValue(&destroyRespPayload, bi.ResponsePayload.(ttlv.TTLV))
```

Comment:
```suggestion
	ttlvPayload, ok := bi.ResponsePayload.(ttlv.TTLV)
	if !ok {
		return "", errors.New("failed to parse responsePayload")
	}

	err = decoder.DecodeValue(&destroyRespPayload, ttlvPayload)
```

add a typecheck here ?
 
(a linter was complaining over at cephcsi)